### PR TITLE
Remove cuhxx files

### PR DESCRIPTION
**Pull request by ArthurCourselle**
Since .cuhxx extension is not conventional they are removed here. Also a little bit refacto on CUDA kernel reduce functions.

### CODE REVIEW

File: Holovibes/cuda_includes/reduce.cuh [hunk 1]
```diff
@@ -1,25 +1,259 @@
...
 
-/*! \brief Reduce add operation
+#include "cuda.h"
+#include "cuda_runtime.h"
+
+#include <cassert>
+#include <limits>
+
+#include "common.cuh"
+#include "cuda_memory.cuh"
```
Comment (Baschtien) [inline Holovibes/cuda_includes/reduce.cuh:16]: Shouldn't the includes be at the beginning of the file ? (after the pragma once)

**Review by Baschtien — APPROVED** [review]

File: Holovibes/cuda_includes/map.cuh [hunk 2]
```diff
@@ -1,87 +1,202 @@
...
+                           func,
+                           stream);
+    else
+        map_generic_caller(output, input, size, func, stream);
+}
 
 /*! \brief Map input (ushort) to output (ushort) throughout a mapping function.
  *
- * \see map_generic float version (above)
+ *  This function is the specialized map operation for float arrays.
```
Comment (Hpn4) [inline Holovibes/cuda_includes/map.cuh:144]: Copy pasted comments. Ushort instead of float

File: Holovibes/cuda_includes/reduce.cuh [hunk 3]
```diff
@@ -1,25 +1,259 @@
...
+
+/*! \brief Mask used by the warp reduce using registers. This mask means every thread should be processed. */
+#define FULL_MASK 0xffffffff
+
+/*! \brief Maximum number of threads in one block. */
+static constexpr uint max_block_size = 1024;
```
Comment (Hpn4) [inline Holovibes/cuda_includes/reduce.cuh:22]: There is a dedicated function in hardware_limits.hh for this

File: Holovibes/cuda_includes/reduce.cuh [hunk 4]
```diff
@@ -1,25 +1,259 @@
...
+    static constexpr I identity_elt_add = static_cast<I>(0);
 
-// Min / Max : ushort not supported by CUDA (because of atomic operation)
+    reduce_generic(output, input, size, op_add, atomic_add, identity_elt_add, stream);
+}
 
-/*! \brief Reduce min operation */
+/*! \brief Reduce min operation.
+ *
+ *  Two types are needed to avoid overflow (sizeof(U) >= sizeof(T))
```
Comment (Hpn4) [inline Holovibes/cuda_includes/reduce.cuh:218]: There is only one type in the function

File: Holovibes/cuda_includes/reduce.cuh [hunk 5]
```diff
@@ -1,25 +1,259 @@
...
+    // The identity element of min operation is the max of the type : min(A, max(T)) = A.
+    static constexpr T identity_elt_min = std::numeric_limits<T>::max();
+
+    reduce_generic(output, input, size, op_min, atomic_min, identity_elt_min, stream);
+}
 
-/*! \brief Reduce max operation */
+/*! \brief Reduce max operation.
+ *
+ *  Two types are needed to avoid overflow (sizeof(U) >= sizeof(T))
```
Comment (Hpn4) [inline Holovibes/cuda_includes/reduce.cuh:239]: Same only one type

**Pull request merged**